awscli/customizations/emr/createdefaultroles.py

```diff
@@ -215,14 +215,15 @@ def _construct_role_and_role_policy_structure(
             list.append({'Role': response['Role'], 'RolePolicy': policy})
             return list
 
-    def _check_if_role_exists(self, role_name, parsed_globals):
```

Comment:
Just wanted to follow up here.  I see the change has been merged and I don't think we need to do anything else here, but I do want to make sure I clarify my thoughts for future changes we may need to make.  I realize I wasn't entirely clear on what I meant.

What I was trying to communicate earlier was that we should only make a method public because we want that to be part of the public interface, not because it makes our tests more "correct" because we're not testing internal methods.  In general for these types of changes I would prefer to just accept the fact that we're testing internal methods and not make it public.  Until such time where we consider rewriting it to be better structured. Consider:

* It's less code (and less risk) than having to find/replace the method name and ensure we change `_foo` to `foo`.  For these types of changes I think as minimal a change as possible is ideal.
* The method really is intended to be internal (i.e if we created another class in this module that was going to use this class, I wouldn't want it calling this internal-but-now-public method).

Yes, avoiding testing internal methods is ideal, but I don't think artificially changing public visibility so we can check the "not-testing-internal-methods" box seems correct.

Hope that helps clarify things going forward.